src/main/java/com/google/devtools/build/lib/remote/TreeNodeRepository.java

```diff
@@ -362,23 +362,21 @@ private synchronized Directory getOrComputeDirectory(TreeNode node) throws IOExc
         TreeNode child = entry.getChild();
         if (child.isLeaf()) {
           ActionInput input = child.getActionInput();
+          final Digest digest;
           if (input instanceof VirtualActionInput) {
             VirtualActionInput virtualInput = (VirtualActionInput) input;
-            Digest digest = digestUtil.compute(virtualInput);
+            digest = digestUtil.compute(virtualInput);
             virtualInputDigestCache.put(virtualInput, digest);
             // There may be multiple inputs with the same digest. In that case, we don't care which
             // one we get back from the digestVirtualInputCache later.
             digestVirtualInputCache.put(digest, virtualInput);
-            b.addFilesBuilder()
-                .setName(entry.getSegment())
-                .setDigest(digest)
-                .setIsExecutable(false);
```

Comment:
Wait, you're changing the virtualInput behavior away from non-executable *to* executable? Aren't these all directories, where the meaning of executable is nil?

Reply:
Good catch!
My brain read what I expected to see, not what was written.

In the solution we talked about, it's supposed to the true, right?

Reply:
Note that this code only applies to leaf nodes. As I understand it a VirtualActionInput is either an empty file or a param file. For the sake of being consistent I think we can also mark it as executable.